core/src/sumeragi/fault.rs

```diff
@@ -630,21 +280,1363 @@ pub fn run_sumeragi_main_loop<F>(
+
+            atomic_sumeragi_guard.latest_block_height = 0;
+
+            assert_eq!(
+                atomic_sumeragi_guard
+                    .current_topology
+                    .role(&sumeragi.peer_id),
+                Role::Leader
+            );
+            let transactions = genesis_network.transactions.clone();
+            {
+                // Don't start genesis round. Instead just commit the genesis block.
+                if transactions.is_empty() {
+                    panic!("Genesis transactions set contains no valid transactions");
+                } else {
+                    let mut wsv_guard = sumeragi.wsv.lock().unwrap();
+                    let block = PendingBlock::new(transactions, Vec::new())
+                        .chain_first_with_genesis_topology(
+                            atomic_sumeragi_guard.current_topology.clone(),
+                        );
+
+                    {
+                        info!(block_hash = %block.hash(), "Publishing genesis block.");
+
+                        let block = block.validate(&sumeragi.transaction_validator, &wsv_guard);
+
+                        info!(
+                            peer_role = ?atomic_sumeragi_guard.current_topology.role(&sumeragi.peer_id),
+                            block_hash = %block.hash(),
+                            "Created a block",
+                        );
+                        for event in Vec::<Event>::from(&block) {
+                            trace!(?event);
+                            drop(sumeragi.events_sender.send(event));
+                        }
+                        let signed_block = block
+                            .sign(sumeragi.key_pair.clone())
+                            .expect("Sign genesis block.");
+                        {
+                            {
+                                /*
+                                TODO: purge the unneeded of these
+                                self.invalidated_blocks_hashes.clear();
+                                self.txs_awaiting_created_block.clear();
+                                self.txs_awaiting_receipts.clear();
+                                self.votes_for_view_change.clear();
+                                */
+                                atomic_sumeragi_guard.latest_block_height =
+                                    signed_block.header().height;
+
+                                sumeragi.broadcast_msg(
+                                    BlockCommitted::from(signed_block.clone()),
+                                    &atomic_sumeragi_guard.current_topology,
+                                );
+
+                                let block = signed_block.commit();
+                                let block_hash = block.hash();
+
+                                if let Err(error) = wsv_guard.apply(block.clone()) {
+                                    panic!("Failed to apply block on WSV. This is absolutely not acceptable.");
```

Comment:
I'd remove the `absolutely not acceptable` remark, or at least rephrase it somehow

Reply:
yea